proxy/envoy/config.go

```diff
@@ -411,6 +413,37 @@ func buildRoutes(svc *model.Service, rulesMap map[string][]*config.RouteRule,
 	return routes
 }
 
+func buildRouteHeaders(headerMatches map[string]*config.StringMatch) []Header {
+	headers := make([]Header, 0, len(headerMatches))
+	for name, stringMatch := range headerMatches {
+		header := buildRouteHeader(name, stringMatch)
+		headers = append(headers, header)
+	}
+	sort.Sort(HeadersByNameValue(headers)) // Guaranteed ordering.
+	return headers
+}
+
+func buildRouteHeader(name string, match *config.StringMatch) Header {
+	value := ""
```

Comment:
It looks like `regex` should default to false? If so, you can take advantage of default values for both string and boolean types, e.g.

var value string
var regex bool